code/xcvm/lib/core/src/packet.rs

```diff
@@ -10,8 +10,8 @@ use serde::{Deserialize, Serialize};
 pub struct XCVMAck(u8);
```

Comment:
Honestly I think we should change it to `enum XCVMAck { Ok, Fail }`.  It’s a bit silly that after decoding a packet we still need to verify value of the ack.  Perhaps in another PR.